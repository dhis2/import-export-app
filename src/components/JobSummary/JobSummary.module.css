.container {
    margin-bottom: 15px;
}

.title {
    font-family: Roboto;
    margin: 0px;
}

.taskDetails {
    margin: 6px 0px 0px;
    font-size: 12px;
    line-height: 14px;
    color: rgb(74, 87, 104);
}

.tags {
    margin-top: 6px;
}

.tags > * {
    margin-right: 6px;
}
