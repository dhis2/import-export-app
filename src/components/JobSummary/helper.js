import PropTypes from 'prop-types'

const statsPropTypeObj = {
    deleted: PropTypes.number.isRequired,
    ignored: PropTypes.number.isRequired,
    total: PropTypes.number.isRequired,
    updated: PropTypes.number.isRequired,
    imported: PropTypes.number,
    created: PropTypes.number,
}

const statsPropType = PropTypes.exact(statsPropTypeObj)

const getClassName = (c) => {
    const s = c.split('.')
    return s[s.length - 1]
}

const typeReportParse = (report) => {
    let allStats = []
    const allMessages = []

    Object.keys(report).forEach((i) => {
        const { klass, objectReports, stats } = report[i]

        allStats = [
            ...allStats,
            {
                ...stats,
                type: getClassName(klass),
            },
        ]

        objectReports &&
            objectReports.forEach((r) => {
                const { uid, errorReports } = r

                errorReports &&
                    errorReports.forEach((e) => {
                        allMessages.push({
                            uid,
                            type: getClassName(e.mainKlass),
                            property: e.errorProperty,
                            message: e.message,
                        })
                    })
            })
    })

    return { stats: allStats, messages: allMessages }
}

const messagesPropType = PropTypes.arrayOf(
    PropTypes.exact({
        uid: PropTypes.string,
        type: PropTypes.string,
        property: PropTypes.string,
        message: PropTypes.string,
    })
)

export { typeReportParse, statsPropType, statsPropTypeObj, messagesPropType }
