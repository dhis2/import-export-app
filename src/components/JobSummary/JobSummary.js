import i18n from '@dhis2/d2-i18n'
import { Divider, Tag } from '@dhis2/ui'
import PropTypes from 'prop-types'
import React from 'react'
import { jsDateToString } from '../../utils/helper'
import { Details } from './Details/Details'
import styles from './JobSummary.module.css'
import { Log } from './Log/Log'
import { Summary } from './Summary/Summary'

const Header = ({ jobDetails, task, showFileDetails }) => (
    <div className={styles.header}>
        <h3 className={styles.title}>{`${i18n.t('Job summary')}`}</h3>
        {showFileDetails && (
            <span className={styles.taskDetails}>
                <span data-test="job-summary-filename">
                    {jobDetails.files[0].name}
                </span>{' '}
                -{' '}
                <span data-test="job-summary-date">
                    {jsDateToString(task.created)}{' '}
                </span>
            </span>
        )}
    </div>
)

Header.propTypes = {
    jobDetails: PropTypes.object.isRequired,
    task: PropTypes.object.isRequired,
    showFileDetails: PropTypes.bool,
}

const Tags = ({ jobDetails, task }) => (
    <div className={styles.tags} data-test="job-summary-tags">
        {task.completed ? (
            <Tag positive>{i18n.t('Completed')}</Tag>
        ) : (
            <Tag neutral>{i18n.t('In progress')}</Tag>
        )}
        {task.error && (
            <Tag negative bold>
                {i18n.t('Error')}
            </Tag>
        )}
        {task.summary &&
            task.summary.conflicts &&
            (task.summary.conflicts.length || null) && (
                <Tag negative>{i18n.t('Conflicts')}</Tag>
            )}
        {jobDetails.dryRun && (
            <Tag neutral bold>
                {i18n.t('Dry run')}
            </Tag>
        )}
    </div>
)

Tags.propTypes = {
    jobDetails: PropTypes.object.isRequired,
    task: PropTypes.object.isRequired,
}

const JobSummary = ({
    task,
    showFileDetails = true,
    showJobDetails = false,
    dataTest,
}) => {
    if (!task) return null
    const { jobDetails } = task

    return (
        <div className={styles.container} data-test={dataTest}>
            <Header
                jobDetails={jobDetails}
                task={task}
                showFileDetails={showFileDetails}
            />
            <Tags jobDetails={jobDetails} task={task} />
            <Divider />
            {task.completed && task.summary && (
                <Summary summary={task.summary} importType={task.importType} />
            )}
            <div className={styles.events}>
                <Log events={task.events} />
            </div>
            {showJobDetails && (
                <div className={styles.jobDetails}>
                    <Details details={task.jobDetails} />
                </div>
            )}
        </div>
    )
}

JobSummary.propTypes = {
    dataTest: PropTypes.string.isRequired,
    showFileDetails: PropTypes.bool,
    showJobDetails: PropTypes.bool,
    task: PropTypes.object,
}

export { JobSummary }
