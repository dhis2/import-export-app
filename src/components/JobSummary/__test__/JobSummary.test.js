import i18n from '@dhis2/d2-i18n'
import React from 'react'
import { render } from 'test-utils'
import '@testing-library/jest-dom/extend-expect'

import { allJobs } from '../../JobOverview/__test__/data'
import { JobSummary } from '../JobSummary'

const props = {
    showFileDetails: true,
    showJobDetails: true,
    dataTest: 'job-summary',
}

const renderJobSummary = (task, props) => {
    const jobSummary = render(<JobSummary task={task} {...props} />)
    return { ...jobSummary }
}

describe('summary for a GML job', () => {
    const gmlJob = allJobs.find((j) => j.importType == 'GML_IMPORT')

    test('type count is showing', () => {
        const { getByDataTest } = renderJobSummary(gmlJob, props)
        const typeCount = getByDataTest('job-summary-type-count')
        expect(typeCount).toBeInTheDocument()
        expect(typeCount).toBeVisible()
    })

    test('messages are showing', () => {
        const { getByDataTest } = renderJobSummary(gmlJob, props)
        const messages = getByDataTest('job-summary-messages')
        expect(messages).toBeInTheDocument()
        expect(messages).toBeVisible()
    })

    test('completed, error and dry run tags are showing', () => {
        const { getByDataTest } = renderJobSummary(gmlJob, props)
        const tags = getByDataTest('job-summary-tags')
        expect(tags).toHaveTextContent(i18n.t('Completed'))
        expect(tags).toHaveTextContent(i18n.t('Error'))
        expect(tags).toHaveTextContent(i18n.t('Dry run'))
    })

    test('events are showing', () => {
        const { getByDataTest } = renderJobSummary(gmlJob, props)
        const events = getByDataTest('job-summary-log')
        expect(events).toBeInTheDocument()
        expect(events).toBeVisible()
    })

    test('job details are showing', () => {
        const { getByDataTest } = renderJobSummary(gmlJob, props)
        const jobDetails = getByDataTest('job-summary-job-details')
        expect(jobDetails).toBeInTheDocument()
        expect(jobDetails).toBeVisible()
    })

    it('matches snapshot', () => {
        const { asFragment } = renderJobSummary(gmlJob, props)
        expect(asFragment()).toMatchSnapshot()
    })
})

describe('summary for a data import job with conflicts', () => {
    const dataJob = allJobs.find(
        (j) => j.importType == 'DATAVALUE_IMPORT' && j.summary.conflicts
    )

    test('overview summary is showing', () => {
        const { getByDataTest } = renderJobSummary(dataJob, props)
        const overviewSummary = getByDataTest('job-summary-single-summary')
        expect(overviewSummary).toHaveTextContent(i18n.t('Summary'))
    })

    test('conflicts are showing', () => {
        const { getByDataTest } = renderJobSummary(dataJob, props)
        const conflicts = getByDataTest('job-summary-conflicts')
        expect(conflicts).toBeInTheDocument()
        expect(conflicts).toBeVisible()
    })

    test('completed, conflicts and dry run tags are showing', () => {
        const { getByDataTest } = renderJobSummary(dataJob, props)
        const tags = getByDataTest('job-summary-tags')
        expect(tags).toHaveTextContent(i18n.t('Completed'))
        expect(tags).toHaveTextContent(i18n.t('Conflicts'))
        expect(tags).toHaveTextContent(i18n.t('Dry run'))
    })

    test('events are showing', () => {
        const { getByDataTest } = renderJobSummary(dataJob, props)
        const events = getByDataTest('job-summary-log')
        expect(events).toBeInTheDocument()
        expect(events).toBeVisible()
    })

    test('job details are showing', () => {
        const { getByDataTest } = renderJobSummary(dataJob, props)
        const jobDetails = getByDataTest('job-summary-job-details')
        expect(jobDetails).toBeInTheDocument()
        expect(jobDetails).toBeVisible()
    })

    it('matches snapshot', () => {
        const { asFragment } = renderJobSummary(dataJob, props)
        expect(asFragment()).toMatchSnapshot()
    })
})
