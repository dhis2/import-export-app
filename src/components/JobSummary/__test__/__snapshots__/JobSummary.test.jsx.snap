// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`summary for a GML job matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
    data-test="job-summary"
  >
    <div
      class="header"
    >
      <h3
        class="title"
      >
        Job summary
      </h3>
      <div
        class="taskDetails"
      >
        <span
          data-test="job-summary-filename"
        >
          gml1.gml
        </span>
         - 
        <span
          data-test="job-summary-date"
        >
          2020-02-12 13:00:32
 
        </span>
      </div>
    </div>
    <div
      class="tags"
      data-test="job-summary-tags"
    >
      <div
        class="jsx-72316749 positive"
        data-test="dhis2-uicore-tag"
      >
        <span
          class="jsx-1110058759"
          data-test="dhis2-uicore-tag-text"
        >
          Completed
        </span>
      </div>
      <div
        class="jsx-72316749 negative bold"
        data-test="dhis2-uicore-tag"
      >
        <span
          class="jsx-1110058759"
          data-test="dhis2-uicore-tag-text"
        >
          Error
        </span>
      </div>
      <div
        class="jsx-72316749 neutral bold"
        data-test="dhis2-uicore-tag"
      >
        <span
          class="jsx-1110058759"
          data-test="dhis2-uicore-tag-text"
        >
          Dry run
        </span>
      </div>
    </div>
    <div
      class="jsx-1555285825 jsx-876534464 "
      data-test="dhis2-uicore-divider"
    />
    <div
      class="typeReportSummary"
      data-test="job-summary-summary"
    >
      <div>
        <div>
          <div
            class="container"
            data-test="job-summary-single-summary"
          >
            <span
              class="label"
            >
              Summary
            </span>
            <table
              class="jsx-2430604489 "
              data-test="dhis2-uicore-table"
            >
              <thead
                data-test="dhis2-uicore-tablehead"
              >
                <tr
                  class="jsx-3124144591 zebraStriping"
                  data-test="dhis2-uicore-tablerowhead"
                >
                  <th
                    class="jsx-1796049005 "
                    data-test="dhis2-uicore-tablecellhead"
                  >
                    Created
                  </th>
                  <th
                    class="jsx-1796049005 "
                    data-test="dhis2-uicore-tablecellhead"
                  >
                    Deleted
                  </th>
                  <th
                    class="jsx-1796049005 "
                    data-test="dhis2-uicore-tablecellhead"
                  >
                    Ignored
                  </th>
                  <th
                    class="jsx-1796049005 "
                    data-test="dhis2-uicore-tablecellhead"
                  >
                    Updated
                  </th>
                  <th
                    class="jsx-1796049005 "
                    data-test="dhis2-uicore-tablecellhead"
                  >
                    Total
                  </th>
                </tr>
              </thead>
              <tbody
                data-test="dhis2-uicore-tablebody"
              >
                <tr
                  class="jsx-3124144591 zebraStriping"
                  data-test="dhis2-uicore-tablerow"
                >
                  <td
                    class="jsx-996330601 "
                    data-test="dhis2-uicore-tablecell"
                  >
                    0
                  </td>
                  <td
                    class="jsx-996330601 "
                    data-test="dhis2-uicore-tablecell"
                  >
                    0
                  </td>
                  <td
                    class="jsx-996330601 "
                    data-test="dhis2-uicore-tablecell"
                  >
                    0
                  </td>
                  <td
                    class="jsx-996330601 "
                    data-test="dhis2-uicore-tablecell"
                  >
                    0
                  </td>
                  <td
                    class="jsx-996330601 "
                    data-test="dhis2-uicore-tablecell"
                  >
                    0
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          class="container"
          data-test="job-summary-type-count"
        >
          <span
            class="label"
          >
            Details by type
          </span>
          <table
            class="jsx-2430604489 "
            data-test="dhis2-uicore-table"
          >
            <thead
              data-test="dhis2-uicore-tablehead"
            >
              <tr
                class="jsx-3124144591 zebraStriping"
                data-test="dhis2-uicore-tablerowhead"
              >
                <th
                  class="jsx-1796049005 "
                  data-test="dhis2-uicore-tablecellhead"
                >
                  Type
                </th>
                <th
                  class="jsx-1796049005 "
                  data-test="dhis2-uicore-tablecellhead"
                >
                  Created
                </th>
                <th
                  class="jsx-1796049005 "
                  data-test="dhis2-uicore-tablecellhead"
                >
                  Deleted
                </th>
                <th
                  class="jsx-1796049005 "
                  data-test="dhis2-uicore-tablecellhead"
                >
                  Ignored
                </th>
                <th
                  class="jsx-1796049005 "
                  data-test="dhis2-uicore-tablecellhead"
                >
                  Updated
                </th>
                <th
                  class="jsx-1796049005 "
                  data-test="dhis2-uicore-tablecellhead"
                >
                  Total
                </th>
              </tr>
            </thead>
            <tbody
              data-test="dhis2-uicore-tablebody"
            >
              <tr
                class="jsx-3124144591 zebraStriping"
                data-test="dhis2-uicore-tablerow"
              >
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  DefaultGmlImportService
                </td>
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  0
                </td>
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  0
                </td>
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  0
                </td>
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  0
                </td>
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  0
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="container"
          data-test="job-summary-messages"
        >
          <span
            class="label"
          >
            Messages
          </span>
          <table
            class="jsx-2430604489 "
            data-test="dhis2-uicore-table"
          >
            <thead
              data-test="dhis2-uicore-tablehead"
            >
              <tr
                class="jsx-3124144591 zebraStriping"
                data-test="dhis2-uicore-tablerowhead"
              >
                <th
                  class="jsx-1796049005 "
                  data-test="dhis2-uicore-tablecellhead"
                >
                  UID
                </th>
                <th
                  class="jsx-1796049005 "
                  data-test="dhis2-uicore-tablecellhead"
                >
                  Type
                </th>
                <th
                  class="jsx-1796049005 "
                  data-test="dhis2-uicore-tablecellhead"
                >
                  Property
                </th>
                <th
                  class="jsx-1796049005 "
                  data-test="dhis2-uicore-tablecellhead"
                >
                  Message
                </th>
              </tr>
            </thead>
            <tbody
              data-test="dhis2-uicore-tablebody"
            >
              <tr
                class="jsx-3124144591 zebraStriping"
                data-test="dhis2-uicore-tablerow"
              >
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                />
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  DefaultGmlImportService
                </td>
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                />
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  Failed to validate job runtime - \`GML import failed: Content is not allowed in prolog. On line 1 column 1.\`
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
    <div
      class="events"
    >
      <div
        class="container"
        data-test="job-summary-log"
      >
        <span
          class="label"
        >
          Log
        </span>
        <table
          class="jsx-2430604489 "
          data-test="dhis2-uicore-table"
        >
          <thead
            data-test="dhis2-uicore-tablehead"
          >
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerowhead"
            >
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Time
              </th>
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Message
              </th>
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                ID
              </th>
            </tr>
          </thead>
          <tbody
            data-test="dhis2-uicore-tablebody"
          >
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerow"
            >
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                2020-02-12 13:00:32

              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                Failed to validate job runtime - \`GML import failed: Content is not allowed in prolog. On line 1 column 1.\`
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                init
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="jobDetails"
    >
      <div
        class="container"
        data-test="job-summary-job-details"
      >
        <span
          class="label"
        >
          Job details
        </span>
        <table
          class="jsx-2430604489 "
          data-test="dhis2-uicore-table"
        >
          <thead
            data-test="dhis2-uicore-tablehead"
          >
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerowhead"
            >
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Key
              </th>
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Value
              </th>
            </tr>
          </thead>
          <tbody
            data-test="dhis2-uicore-tablebody"
          >
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerow"
            >
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                files
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                gml1.gml
              </td>
            </tr>
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerow"
            >
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                dryRun
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                true
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`summary for a data import job with conflicts matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
    data-test="job-summary"
  >
    <div
      class="header"
    >
      <h3
        class="title"
      >
        Job summary
      </h3>
      <div
        class="taskDetails"
      >
        <span
          data-test="job-summary-filename"
        >
          data2.json
        </span>
         - 
        <span
          data-test="job-summary-date"
        >
          2020-02-12 13:00:59
 
        </span>
      </div>
    </div>
    <div
      class="tags"
      data-test="job-summary-tags"
    >
      <div
        class="jsx-72316749 positive"
        data-test="dhis2-uicore-tag"
      >
        <span
          class="jsx-1110058759"
          data-test="dhis2-uicore-tag-text"
        >
          Completed
        </span>
      </div>
      <div
        class="jsx-72316749 negative"
        data-test="dhis2-uicore-tag"
      >
        <span
          class="jsx-1110058759"
          data-test="dhis2-uicore-tag-text"
        >
          Conflicts
        </span>
      </div>
      <div
        class="jsx-72316749 neutral bold"
        data-test="dhis2-uicore-tag"
      >
        <span
          class="jsx-1110058759"
          data-test="dhis2-uicore-tag-text"
        >
          Dry run
        </span>
      </div>
    </div>
    <div
      class="jsx-1555285825 jsx-876534464 "
      data-test="dhis2-uicore-divider"
    />
    <div
      class="container"
      data-test="job-summary-summary"
    >
      <div>
        <div
          class="container"
          data-test="job-summary-single-summary"
        >
          <span
            class="label"
          >
            Summary
          </span>
          <table
            class="jsx-2430604489 "
            data-test="dhis2-uicore-table"
          >
            <thead
              data-test="dhis2-uicore-tablehead"
            >
              <tr
                class="jsx-3124144591 zebraStriping"
                data-test="dhis2-uicore-tablerowhead"
              >
                <th
                  class="jsx-1796049005 "
                  data-test="dhis2-uicore-tablecellhead"
                >
                  Status
                </th>
                <th
                  class="jsx-1796049005 "
                  data-test="dhis2-uicore-tablecellhead"
                >
                  Description
                </th>
              </tr>
            </thead>
            <tbody
              data-test="dhis2-uicore-tablebody"
            >
              <tr
                class="jsx-3124144591 zebraStriping"
                data-test="dhis2-uicore-tablerow"
              >
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  WARNING
                </td>
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  Import process completed successfully
                </td>
              </tr>
            </tbody>
          </table>
          <table
            class="jsx-2430604489 "
            data-test="dhis2-uicore-table"
          >
            <thead
              data-test="dhis2-uicore-tablehead"
            >
              <tr
                class="jsx-3124144591 zebraStriping"
                data-test="dhis2-uicore-tablerowhead"
              >
                <th
                  class="jsx-1796049005 "
                  data-test="dhis2-uicore-tablecellhead"
                >
                  Created
                </th>
                <th
                  class="jsx-1796049005 "
                  data-test="dhis2-uicore-tablecellhead"
                >
                  Deleted
                </th>
                <th
                  class="jsx-1796049005 "
                  data-test="dhis2-uicore-tablecellhead"
                >
                  Ignored
                </th>
                <th
                  class="jsx-1796049005 "
                  data-test="dhis2-uicore-tablecellhead"
                >
                  Updated
                </th>
                <th
                  class="jsx-1796049005 "
                  data-test="dhis2-uicore-tablecellhead"
                >
                  Total
                </th>
              </tr>
            </thead>
            <tbody
              data-test="dhis2-uicore-tablebody"
            >
              <tr
                class="jsx-3124144591 zebraStriping"
                data-test="dhis2-uicore-tablerow"
              >
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  0
                </td>
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  0
                </td>
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  173
                </td>
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  0
                </td>
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  173
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="container"
          data-test="job-summary-conflicts"
        >
          <span
            class="label"
          >
            Conflicts
          </span>
          <table
            class="jsx-2430604489 "
            data-test="dhis2-uicore-table"
          >
            <thead
              data-test="dhis2-uicore-tablehead"
            >
              <tr
                class="jsx-3124144591 zebraStriping"
                data-test="dhis2-uicore-tablerowhead"
              >
                <th
                  class="jsx-1796049005 "
                  data-test="dhis2-uicore-tablecellhead"
                >
                  Object
                </th>
                <th
                  class="jsx-1796049005 "
                  data-test="dhis2-uicore-tablecellhead"
                >
                  Value
                </th>
              </tr>
            </thead>
            <tbody
              data-test="dhis2-uicore-tablebody"
            >
              <tr
                class="jsx-3124144591 zebraStriping"
                data-test="dhis2-uicore-tablerow"
              >
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  V6L425pT3A0
                </td>
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  Category option combo not found or not accessible for writing data
                </td>
              </tr>
              <tr
                class="jsx-3124144591 zebraStriping"
                data-test="dhis2-uicore-tablerow"
              >
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  Prlt0C1RF0s
                </td>
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  Category option combo not found or not accessible for writing data
                </td>
              </tr>
              <tr
                class="jsx-3124144591 zebraStriping"
                data-test="dhis2-uicore-tablerow"
              >
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  hEFKSsPV5et
                </td>
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  Category option combo not found or not accessible for writing data
                </td>
              </tr>
              <tr
                class="jsx-3124144591 zebraStriping"
                data-test="dhis2-uicore-tablerow"
              >
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  psbwp3CQEhs
                </td>
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  Category option combo not found or not accessible for writing data
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        class="rest"
        data-test="job-summary-summary-rest"
      />
    </div>
    <div
      class="events"
    >
      <div
        class="container"
        data-test="job-summary-log"
      >
        <span
          class="label"
        >
          Log
        </span>
        <table
          class="jsx-2430604489 "
          data-test="dhis2-uicore-table"
        >
          <thead
            data-test="dhis2-uicore-tablehead"
          >
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerowhead"
            >
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Time
              </th>
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Message
              </th>
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                ID
              </th>
            </tr>
          </thead>
          <tbody
            data-test="dhis2-uicore-tablebody"
          >
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerow"
            >
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                2020-02-12 12:00:58

              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                Initiated dataValueImport
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                init
              </td>
            </tr>
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerow"
            >
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                2020-02-12 12:00:59

              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                Process started
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                oE17j9gw54i
              </td>
            </tr>
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerow"
            >
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                2020-02-12 12:00:59

              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                Importing data values
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                SKEyN7hdwyd
              </td>
            </tr>
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerow"
            >
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                2020-02-12 12:00:59

              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                Import done
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                CO2RDYqaGde
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="jobDetails"
    >
      <div
        class="container"
        data-test="job-summary-job-details"
      >
        <span
          class="label"
        >
          Job details
        </span>
        <table
          class="jsx-2430604489 "
          data-test="dhis2-uicore-table"
        >
          <thead
            data-test="dhis2-uicore-tablehead"
          >
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerowhead"
            >
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Key
              </th>
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Value
              </th>
            </tr>
          </thead>
          <tbody
            data-test="dhis2-uicore-tablebody"
          >
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerow"
            >
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                files
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                data2.json
              </td>
            </tr>
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerow"
            >
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                format
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                JSON
              </td>
            </tr>
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerow"
            >
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                dryRun
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                true
              </td>
            </tr>
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerow"
            >
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                strategy
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                Updates only
              </td>
            </tr>
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerow"
            >
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                preheatCache
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                false
              </td>
            </tr>
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerow"
            >
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                skipAudit
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                false
              </td>
            </tr>
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerow"
            >
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                dataElementIdScheme
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                Uid
              </td>
            </tr>
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerow"
            >
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                orgUnitIdScheme
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                Uid
              </td>
            </tr>
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerow"
            >
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                idScheme
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                Uid
              </td>
            </tr>
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerow"
            >
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                skipExistingCheck
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                true
              </td>
            </tr>
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerow"
            >
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                firstRowIsHeader
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                false
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
